editoast/src/views/infra/routes.rs

```diff
@@ -119,3 +128,78 @@ async fn get_routes_track_ranges<'a>(
 
     Ok(Json(result))
 }
+
+#[cfg(test)]
+mod tests {
+    use actix_http::StatusCode;
+    use actix_web::test::{call_service, read_body_json, TestRequest};
+    use rstest::rstest;
+
+    use crate::{
+        fixtures::tests::{db_pool, small_infra},
+        views::{
+            infra::routes::{WaypointResponse, WaypointType},
+            tests::create_test_service,
+        },
+    };
+
+    #[rstest]
+    async fn get_waypoint_should_return_da2_detector() {
+        let app = create_test_service().await;
+        let small_infra = small_infra(db_pool()).await;
+        let small_infra_id = small_infra.id();
+        let waypoint_type = WaypointType::Detector;
+        let request = TestRequest::get()
+            .uri(format!("/infra/{small_infra_id}/routes/{waypoint_type}/DA2").as_str())
+            .to_request();
+        let response = call_service(&app, request).await;
+        assert_eq!(response.status(), StatusCode::OK);
+        let waypoint: WaypointResponse = read_body_json(response).await;
+        assert_eq!(
+            waypoint,
+            WaypointResponse::new(
+                vec!["rt.DA2->DA5".to_string()],
+                vec!["rt.buffer_stop.0->DA2".to_string()]
+            )
+        );
+    }
+
+    #[rstest]
+    async fn get_waypoint_should_return_buffer_stop_number_4() {
```

Comment:
Not sure about the naming, this should not return a detector, but a list of routes from a detector.
Also I'm not a big fan of naming test function with an id inside. (here number 4)

Reply:
Done.